design-docs/improved-console.md

```diff
@@ -0,0 +1,422 @@
+# Improved Console Output
+
+## Primary Goals
+* Improve perceived performance by making user aware of background work
+* Modern feel
+* User extensibility
```

Comment:
This is the difference between [plain output](https://gradle.slack.com/files/eric/F3HQSGH7B/console-plain.gif) and [rich output](https://gradle.slack.com/files/eric/F3JHLNA6Q/console-design-mockup-failure.gif) — Given the current spec, I believe that if we want to achieve the latter (and I think we do) it is very little additional effort to allow users to do so. Node and Python tools have achieved a sense of modernism by including pretty (colorized, non-ASCII) output in their tools, and I think we should do the same.

Here the reasons I think that could be beneficial:
 * It gives users a new vehicle for contribution for Gradle. I believe someone out there has some crazy awesome ideas for the console UI
 * It gives users an easy way for colorized output to match the rest of their Terminal.